gtsfm/loader/hilti_loader.py

```diff
@@ -233,53 +255,34 @@ def get_relative_pose_prior(self, i1: int, i2: int) -> Optional[PosePrior]:
         return None
 
     def get_absolute_pose_prior(self, idx: int) -> Optional[PosePrior]:
-        rig_idx: int = self.map_image_idx_to_rig(idx)
-        cam_idx: int = self.map_index_to_camera(idx)
+        rig_idx: int = self.rig_from_image(idx)
+        cam_idx: int = self.camera_from_image(idx)
 
         if rig_idx in self._w_T_imu:
             w_T_cam = self._w_T_imu[rig_idx] * self._cam_T_imu_poses[cam_idx].inverse()
             return PosePrior(value=w_T_cam, covariance=None, type=PosePriorType.SOFT_CONSTRAINT)
 
         return None
 
-    def is_valid_pair(self, idx1: int, idx2: int) -> bool:
-        """Checks if (idx1, idx2) is a valid pair. idx1 < idx2 is required.
-
-        Args:
-            idx1: first index of the pair.
-            idx2: second index of the pair.
-
-        Returns:
-            validation result.
-        """
-        if not super().is_valid_pair(idx1, idx2):
-            return False
-
-        rig_idx_i1 = self.map_image_idx_to_rig(idx1)
-        rig_idx_i2 = self.map_image_idx_to_rig(idx2)
-
-        cam_idx_i1 = self.map_index_to_camera(idx1)
-        cam_idx_i2 = self.map_index_to_camera(idx2)
-        if rig_idx_i1 == rig_idx_i2:
-            return (cam_idx_i1, cam_idx_i2) in INTRA_RIG_VALID_PAIRS
-        elif rig_idx_i1 < rig_idx_i2 and rig_idx_i2 - rig_idx_i1 <= self._max_frame_lookahead:
-            return (cam_idx_i1, cam_idx_i2) in INTER_RIG_VALID_PAIRS
-
-    def map_index_to_camera(self, index: int) -> int:
+    def camera_from_image(self, index: int) -> int:
+        """Map image index to camera-on-rig index."""
         return index % NUM_CAMS
 
-    def map_image_idx_to_rig(self, index: int) -> int:
+    def rig_from_image(self, index: int) -> int:
+        """Map image index to rig index."""
         return index // NUM_CAMS
 
+    def image_from_rig_and_camera(self, rig_index: int, camera_idx: int) -> int:
+        """Map image index to rig index."""
+        return rig_index * NUM_CAMS + camera_idx
+
     def get_relative_pose_priors(self, pairs: List[Tuple[int, int]]) -> Dict[Tuple[int, int], PosePrior]:
         pairs = set(pairs)
-        # just add all possible pairs which belong to the same rig (as it will have hard relative prior)
-        for i in range(len(self)):
-            for j in range(i + 1, i + NUM_CAMS - 1):
-                if self.map_image_idx_to_rig(i) == self.map_image_idx_to_rig(j):
-                    pairs.add((i, j))
-                else:
-                    break
+        # For every rig index, add a "star" from camera 2 to 0,1,3,4:
+        for rig_index in range(self.max_rig_index):
+            camera_2 = self.image_from_rig_and_camera(rig_index, 2)
+            for cam_idx in [0, 1, 3, 4]:
+                pairs.add((camera_2, self.image_from_rig_and_camera(rig_index, cam_idx)))
```

Comment:
Can we please add (i1, i2) s.t. i1 < i2? This is the scheme we have followed till now and I am worried there might be some assumption somewhere.

Reply:
Done